packages/fx-core/src/plugins/resource/cicd/providers/provider.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { FxError, Result, ok } from "@microsoft/teamsfx-api";
+import * as fs from "fs-extra";
+import { FileSystemError, InternalError, NoProjectOpenedError } from "../errors";
+import { TemplateKind } from "./enums";
+import path from "path";
+import Mustache from "mustache";
+import { getTemplatesFolder } from "../../../..";
+
+export class CICDProvider {
+  public scaffoldTo = "";
+  public providerName = "";
+  public sourceTemplateName?: (templateName: string) => string;
+  public targetTemplateName?: (templateName: string, envName: string) => string;
+  public async scaffold(
+    projectPath: string,
+    templateName: string,
+    replacements: any
+  ): Promise<Result<boolean, FxError>> {
+    // 0. Preconditions check.
+    if (!(await fs.pathExists(projectPath))) {
+      throw new NoProjectOpenedError();
+    }
+    if (!Object.values<string>(TemplateKind).includes(templateName)) {
+      throw new InternalError(`${templateName} as template kind was not recognized.`);
+    }
+    if (!this.sourceTemplateName || !this.targetTemplateName) {
+      throw new InternalError("sourceTemplateName or targetTemplateName shoudn't be undefined.");
+    }
+
+    // 1. Ensure the target path is existing.
+    const targetPath = path.join(projectPath, this.scaffoldTo);
+    try {
+      await fs.ensureDir(targetPath);
+    } catch (e) {
+      throw new FileSystemError(`Fail to create path: ${targetPath}`, e as Error);
+    }
+
+    // 2. Read README from local.
```

Comment:
Maybe the function name could be "generate README file" which is more customer facing

Reply:
I guess you mean the comment since there's no function in this code snippet.